### PR TITLE
Enable cronet compression

### PR DESCRIPTION
**Pull request by muxi**
Allow Cronet transport to send and receive compressed payloads.

Cherry-picked a bug fix (#10208) that allows chttp2 to send :status header field in trailer-only response.

### CODE REVIEW

File: src/core/ext/transport/chttp2/transport/chttp2_transport.c [hunk 1]
```diff
@@ -1775,6 +1776,26 @@ static void close_from_api(grpc_exec_ctx *exec_ctx, grpc_chttp2_transport *t,
      It's complicated by the fact that our send machinery would be dead by
      the time we got around to sending this, so instead we ignore HPACK
      compression and just write the uncompressed bytes onto the wire. */
+  if (!s->sent_initial_metadata) {
```
Comment (makdharma) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.c]: Please comment the thinking behind this condition?

File: src/core/ext/transport/cronet/transport/cronet_transport.c [hunk 2]
```diff
@@ -643,7 +655,7 @@ static void create_grpc_frame(grpc_slice_buffer *write_slice_buffer,
   *pp_write_buffer = write_buffer;
   uint8_t *p = (uint8_t *)write_buffer;
   /* Append 5 byte header */
-  *p++ = 0;
+  *p++ = (flags & GRPC_WRITE_INTERNAL_COMPRESS) ? 1 : 0;
```
Comment (makdharma) [inline src/core/ext/transport/cronet/transport/cronet_transport.c]: please comment why this byte is special?

File: src/core/ext/transport/chttp2/transport/chttp2_transport.c (hunk 1 above)
Comment (muxi) [inline src/core/ext/transport/chttp2/transport/chttp2_transport.c]: Cherry picked from another PR. Will comment there.

**Comment by muxi** [comment]
Test failures:
Basic Tests Linux - #9674 
Basic Tests Mac - #10138   Manual test for a few hours succeeded

**Pull request merged**